### PR TITLE
executor, tests: move test cases from executor to integrationtest (PART 3)

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: ref #47076

Problem Summary: update `compact_table`, `adapter`, `cte`, `explain`, `explainfor`, `grant`. Total case number is 55.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/integrationtest/t/executor/grant.test [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+DROP USER IF EXISTS 'test'@'%';
+# Grant without a password.
+GRANT ALL PRIVILEGES ON *.* to 'test'@'%';
+# Make sure user is created automatically when grant to a non-exists one.
+SELECT user, plugin FROM mysql.user WHERE user='test' and host='%';
+DROP USER IF EXISTS 'test'@'%';
+set sql_mode=default;
+
+# TestCreateUserWithTooLongName
+-- error 1470
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/grant.test:47]: Do we need to drop user first? Will this test be run repeatedly?

File: tests/integrationtest/t/executor/explainfor.test [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+select @@last_plan_from_cache;
+--sorted_result
+execute stmt using @a, @b, @a;
+prepare stmt from 'with recursive c(p) as (select ?), cte(a, b) as (select 1, 1 union select a+?, 1 from cte, c where a < ?)  select * from cte order by 1, 2;';
+show warnings;
+--sorted_result
+execute stmt using @a, @a, @e;
+--sorted_result
+execute stmt using @b, @b, @c;
+select @@last_plan_from_cache;
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/explainfor.test:225]:
```suggestion
select @@last_plan_from_cache;
set tidb_enable_prepared_plan_cache=default;
set @@tidb_enable_collect_execution_info=default;
```

File: tests/integrationtest/t/executor/cte.test [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+with recursive cte1(c1) as (select c1 from t1 union all select c1 + 1 from cte1 limit 2 offset 4) select * from cte1;
+with recursive cte1(c1) as (select c1 from t1 union all select c1 + 1 from cte1 limit 3 offset 4) select * from cte1;
+with recursive cte1(c1) as (select c1 from t1 union all select c1 + 1 from cte1 limit 4 offset 4) select * from cte1;
+set cte_max_recursion_depth = default;
+
+# TestCTEsInView
+#https://github.com/pingcap/tidb/issues/33965
+create database if not exists executor__cte1;
+create table executor__cte.t (a int);
+create table executor__cte1.t (a int);
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/cte.test:137]: drop tables first?

File: tests/integrationtest/t/executor/cluster_table.test [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+select tidb_decode_sql_digests('["e6f07d43b5c21db0fbb9a31feac2dc599787763393dd5acbfad80e247eb02ad5",1,null,"58f3717da2d79c14773a1e3094aaddeff2b11747d3aef95741151af9acba9d44",{"a":1},[2],"7e9f826ed22a09940940a42d1aca47a75b3adc3c3fde252f4b912ac886194eb9","","abcde"]');
+select tidb_decode_sql_digests('{"a":1}');
+show warnings;
+select tidb_decode_sql_digests('aabbccdd');
+show warnings;
+-- error 1582
+select tidb_decode_sql_digests('a', 1, 2);
+-- error 1582
+select tidb_decode_sql_digests();
+set global tidb_enable_stmt_summary = 0;
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/cluster_table.test:54]:
```suggestion
set global tidb_enable_stmt_summary = default;
```

File: tests/integrationtest/t/executor/grant.test (hunk 1 above)
Comment (Defined2014) [reply tests/integrationtest/t/executor/grant.test:47]: Drop user will report error, the user name is too long.

Comment (SeaRise) [reply tests/integrationtest/t/executor/grant.test:47]: ok

**Review by SeaRise — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**